homeassistant/components/stream/worker.py

```diff
@@ -535,3 +538,9 @@ def is_video(packet: av.Packet) -> Any:
                 raise StreamWorkerError("Error demuxing stream: %s" % str(ex)) from ex
 
             muxer.mux_packet(packet)
+
+            if packet.is_keyframe and packet.stream.type == "video":
+                keyframe_converter.packet = packet
+
+            if keyframe_converter.image_requested:
+                keyframe_converter.check_generate_image()
```

Comment:
One other design idea -- if you were ok to compromise on re-using decoded results --  could be to a local variable here outside of the loop that is "last_keframe" that gets updated on line 543, then here call `generate_image(last_keyframe)`. The difference would be less state inside of keyframe converter.

Reply:
I think we should always reuse decoded/encoded results and avoid doing decoding or encoding twice. These operations are (relatively) expensive.

Reply:
Sounds good. thanks for considering.